source/usage-examples/transaction-conv.txt

```diff
@@ -0,0 +1,188 @@
+- The system went offline
+
+You can perform a transaction to ensure you avoid
+making any partial updates if any single operation in the transaction
+fails.
+
+Sample Data
+~~~~~~~~~~~
+
+Before any transactions, the ``inventory`` collection contains the
+following documents:
+
+.. literalinclude:: /code-snippets/transactions/txn-convenient.js
+   :language: javascript
+   :dedent:
+   :copyable: false
+   :start-after: start-inventory
+   :end-before: end-inventory
+
+You store purchase records in the ``orders`` collection of the
+``testdb`` database. This collection is empty, as there have been no
+purchases.
+
+Implementation
+~~~~~~~~~~~~~~
+
+The code example in this section demonstrates how to use the Convenient
+Transaction API to perform a multi-document transaction in a session. In
+this example, the transaction reflects the actions needed when a
+customer purchases items from your store.
+
+This example code performs a transaction through the following actions:
+
+1. Calls the ``withSession()`` method on the client to implicitly create
+   the session and include the transaction within this session.
+
+#. Calls the ``withTransaction()`` method on the session to create a
+   transaction, run the specified operations, and commit the
+   transaction. If the transaction fails, this method ends the
+   transaction and returns an error message.
+
+#. Performs the following operations within the transaction:
+
+   * To process a predefined ``order`` variable that specifies the item and
+     quantity that the customer wants, the first operation checks if
+     there is sufficient inventory to fulfill the purchase.
+   * If the items are not available or there are not enough items, the
+     transaction ends and the function returns an error message.
+   * If the inventory is sufficient for the order, the code performs an
+     update operation in the ``inventory`` collection to reduce the
+     quantities of the items purchased.
+   * The code inserts a record of the purchase to the ``orders``
+     collection that includes a timestamp, a list of the purchased
+     items, and a total price for the purchase.
+   * The code returns a message acknowledging that the transaction
+     committed successfully with a copy of the purchase record.
+
+#. Prints the return type of ``withSession()``, which is either the
+   error message or the acknowledgment that the transaction completed.
+
+.. literalinclude:: /code-snippets/transactions/txn-convenient.js
+   :language: javascript
+   :linenos:
+   :dedent:
+   :start-after: start-transaction
+   :end-before: end-transaction
+
+Sample Orders and Transaction Results
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When your application completes the payment transaction, the collections
+in the database contains the inventory and order updates. 
+If an error interrupts the transaction, none of the changes appear in
+your database.
+
+Suppose a customer sends the following order:
```

Comment:
Suggestion:
I think it would be easier to scan if the reader could identify this quickly a "transaction completed" example and the other code snippet was identifiable as a "transaction rolled back" example. E.g.

"The following sample order shows a successful order in which sufficient inventory exists and the transaction successfully completes:"

"The following sample order shows an unsuccessful order in which insufficient inventory exists and the transaction ends and rolls back the changes:"